### DISCUSSION TITLE
[Feature] Update exif in asset-viewer

### DISCUSSION THREAD

**Discussion by flyotlin** [immich-app/immich#3818] [Feature Request]
### The feature

Currently users cannot update/adjust exif information of assets (images) except for description. It would be much more convenient if **users can adjust exif in asset-viewer on web**.

I've roughly traced how immich saves uploaded images (`/api/asset/upload`), and found there's an existing asset update api (`/api/asset/:id`), on which I think we can build this new feature.

When assets get uploaded, it seems immich would extract asset's exif information, and saves them into db (`immich.assets`, `immich.exif`). So, updating exif should only occur on db's information instead of asset itself, I suppose.

Since there's many tags in exif, I would recommend start from **date** and **geo** first, and if this feature is feasible, I can help with server-side and web-side code.

By the way, any ideas are welcome!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
I'm closing this as a duplicate of #1696. As discussed in there, it would be ideal if edited metadata got written back to an XMP sidecar file, so that it doesn't get overwritten on later metadata extraction runs. 

I think a PR for this feature would be welcome, and a small start like you mention would be a good idea. Maybe just support for editing the date and writing that change to a sidecar file, with room to add support for more fields later.

**Comment by jmathai** [comment]
I wanted to write exif back to my photos and videos (not sidecars). I created [immich-exif](https://github.com/jmathai/immich-exif) which you can simply add to your docker-compose.yaml file.

Using this until it’s supported natively :).
